### ISSUE TITLE
Unable to find an element using a class name

### ISSUE THREAD

**Issue by PriyoBh** [DevExpress/testcafe#2536]
### Are you requesting a feature or reporting a bug?
Bug (although this might be an issue with my code)

### What is the current behavior?
i am trying to load an element after the page gets loaded

### What is the expected behavior?
After the page gets loaded I expect to find the element

### How would you reproduce the current behavior (if this is a bug)?
Just run the following code.

#### Provide the test code and the tested page URL (if applicable)

Tested page URL:
https://stackla-testing.s3.amazonaws.com/automation/prod/Slider.html
Test code

```js
import { Selector } from "testcafe";

fixture`Test all the widgets gets loaded successfully`
  .page`http://www.google.com`;

//const sliderPage = new ExternalWidget();

test.page`https://stackla-testing.s3.amazonaws.com/automation/prod/Slider.html`(
  "Test whether a slider widget gets loaded",
  async t => {
    const widgetParagraphElement = await Selector(".container").with({
      visibilityCheck: true
    })();
    const container = Selector(".container").exists;
    await t.expect(container).ok();
  }
);
```

### Specify your

* operating system: mac high sierra
* testcafe version:0.20.2
* node.js version:9.11.1

**Comment by josephj** [comment]
@PriyoBh 
The selector you mentioned is inside the the dynamic generated iframe. So you have to wait and switch context. 
http://devexpress.github.io/testcafe/documentation/test-api/working-with-iframes.html

**Comment by miherlosev** [comment]
Yes, @josephj is right.

You should change your code as follows:
```js
test.page`https://stackla-testing.s3.amazonaws.com/automation/prod/Slider.html`(
  "Test whether a slider widget gets loaded",
  async t => {
	await t.switchToIframe('iframe');    
	  
    const container = Selector(".container").exists;
    await t.expect(container).ok();
  }
);
```

**Comment by PriyoBh** [comment]
Thanks - it worked. It was my mistake - for not inspecting the DOM properly. I am a but curious I thought this line from my code should have generated an exception.
```const widgetParagraphElement = await Selector(".container").with({
      visibilityCheck: true
    })();```
As per our documentation on this page - https://devexpress.github.io/testcafe/documentation/test-api/selecting-page-elements/selectors/selector-options.html#optionsvisibilitycheck

its mentioned that - _If the target element is not visible, the selector throws an exception in all these cases._

what I noticed is after execution on that line - widgetParagraphElement had null value
Can you please clarify? Thanks again for your prompt response.

**Comment by PriyoBh** [comment]
Thanks - it worked. It was my mistake - for not inspecting the DOM properly. I am a but curious I thought this line from my code should have generated an exception.

const widgetParagraphElement = await Selector(".container").with({
      visibilityCheck: true
    })();

Instead the variable widgetParagraphElement had a null value.

based on your documentation - https://devexpress.github.io/testcafe/documentation/test-api/selecting-page-elements/selectors/selector-options.html#optionsvisibilitycheck

it says there - "If the target element is not visible, the selector throws an exception in all these cases."

**Comment by rpolishetti** [comment]
I got this issue. Not throwing exception, instead testing will be continued with null value

**Comment by rpolishetti** [comment]
@AndreyBelym 

Got your point about exceptions. But I expected error as per documents, it throws error after timeout specified with Selector API options,  when we try to select DOM element. Instead I observed like, test just continues. When I wait for visible property on Selector promise, I got the real waiting for element. I just used like below.

const elToWait = await Selector('#elementId', {timeout: 20000, visibilityCheck: true}).visible;

With this step it is working as expected.
